File: content/blocks/go/top-words.go
func topWords(counts map[string]int, k int) []string {
	words := slices.Collect(maps.Keys(counts))
	slices.SortFunc(words, func(a, b string) int {
		if c := cmp.Compare(counts[b], counts[a]); c != 0 {
			return c
		}
		return strings.Compare(a, b)
	})
	if len(words) > k {
		words = words[:k]
	}
	return words
}

File: content/blocks/python/top-words.py
def top_words(text, k=3):
    words = re.findall(r"[a-z']+", text.lower())
    counts = collections.Counter(words)
    for word, count in counts.most_common(k):
        print(f"{word:<12} {count:>4}")
    return counts.most_common(k)
